[Bug]: cmake - WOLFSSL_LEAN_TLS broken

### Contact Details

code@jrolli.dev

### Version

5.7.6-stable

### Description

Using `WOLFSSL_LEAN_TLS` causes a transient compilations issue when `WOLFSSL_MEMORY=OFF` is not provided.  The compilation will initially fail with the error given in the logs section, but touching any file that would trigger recompilation of WolfSSL will cause subsequent compilations to succeed.  At first glance, `CMakeLists.txt:1133` appears to be a bad assumption as the only difference between runs is that `WOLFSSL_MEMORY=OFF` is in the cache.  Both removing the conditional as well as adding the `list` command in place of this line fix the issue.

### Reproduction steps

1. Create a new directory with the attached files (`CMakeLists.txt` and save C code below as "repro.c" - cannot upload due to GitHub restrictions...)

```c
#include "wolfssl/options.h"
#include "wolfssl/wolfssl.h"

#include <sys/socket.h>

int
main (void)
{
    const SSL_METHOD *method = wolfTLS_client_method();
    SSL_CTX          *ctx    = wolfSSL_CTX_new(method);
    wolfSSL_CTX_free(ctx);
    return EXIT_FAILURE;
}
```

2. Run `cmake -B build -DWOLFSSL_LEAN_TLS=ON .`
3. Run `cmake --build build`

[CMakeLists.txt](https://github.com/user-attachments/files/18805786/CMakeLists.txt)

### Relevant log output

```shell
Undefined symbols for architecture arm64:
  "_wolfSSL_Free", referenced from:
      _wc_HashNew in hash.c.o
      _wc_HashDelete in hash.c.o
      _wc_rng_new in random.c.o
      __InitRng in random.c.o
      _wc_rng_new_ex in random.c.o
      _wc_rng_free in random.c.o
      _wc_FreeRng in random.c.o
      ...
  "_wolfSSL_Malloc", referenced from:
      _wc_HashNew in hash.c.o
      _wc_rng_new in random.c.o
      __InitRng in random.c.o
      _wc_rng_new_ex in random.c.o
      __sp_invmod_mont_ct in sp_int.c.o
      __sp_exptmod_nct in sp_int.c.o
      _wc_AesNew in aes.c.o
      ...
  "_wolfSSL_Realloc", referenced from:
      _wolfSSL_EVP_CipherUpdate_GCM in ssl.c.o
      _wolfSSL_EVP_PKEY_CTX_add1_hkdf_info in ssl.c.o
      _ECC_populate_EVP_PKEY in ssl.c.o
      _wolfssl_asn1_bit_string_grow in ssl.c.o
      _wolfSSL_BIO_set_conn_hostname in ssl.c.o
      _wolfSSL_BUF_MEM_resize in ssl.c.o
      _wolfSSL_X509_set_ext in ssl.c.o
      ...
ld: symbol(s) not found for architecture arm64
cc: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [_deps/wolfssl-build/libwolfssl.43.0.0.dylib] Error 1
make[1]: *** [_deps/wolfssl-build/CMakeFiles/wolfssl.dir/all] Error 2
make: *** [all] Error 2
```

--- a/.github/workflows/cmake.yml
+++ b/.github/workflows/cmake.yml
@@ -83,5 +83,25 @@ jobs:
         cmake --build .
         cmake --install .
 
+        # clean up
+        cd ..
+        rm -rf build
+
         # Kyber Cmake broken
         # -DWOLFSSL_KYBER:BOOL=yes
+
+# build "lean-tls" wolfssl
+    - name: Build wolfssl with lean-tls
+      working-directory: ./wolfssl
+      run: |
+        mkdir build
+        cd build
+        cmake -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DWOLFSSL_INSTALL=yes -DCMAKE_INSTALL_PREFIX="$GITHUB_WORKSPACE/install" \
+            -DWOLFSSL_LEAN_TLS:BOOL=yes \
+            ..
+        cmake --build .
+        cmake --install .
+
+        # clean up
+        cd ..
+        rm -rf build

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1141,8 +1141,7 @@ if(NOT WOLFSSL_MEMORY)
 else()
     # turn off memory cb if leanpsk or leantls on
     if(WOLFSSL_LEAN_PSK OR WOLFSSL_LEAN_TLS)
-        # but don't turn on NO_WOLFSSL_MEMORY because using own
-        override_cache(WOLFSSL_MEMORY "no")
+        list(APPEND WOLFSSL_DEFINITIONS "-DNO_WOLFSSL_MEMORY")
     endif()
 endif()
 

